### PR TITLE
UI changes to enable GoogleLogin

### PR DESCRIPTION
**Pull request by mahimavuppuluri**
1. Added a Home Page
2. Added user page: only ADMIN users can use this page, to create other users. ADMIN users need to be created by manually adding DB entry
3. Added Login button to AppInBar
4. Login button changes to google avatar after successful Login
5. Clicking avatar shows Logout button
6. Every call to backend goes with Auth Token now

1, 3
Screenshot 2021-01-17 at 11 26 56 PM

2
Screenshot 2021-01-17 at 11 27 36 PM
 

4, 5
Screenshot 2021-01-17 at 11 27 20 PM

Screenshot 2021-01-17 at 11 45 21 PM

### CODE REVIEW

File: src/Pages/User.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  };
+
+  const handleSubmit = () => {
+    const payload = {
+      email: email,
+    };
+
+    createUser(payload)
+      .then((_response) => {
+        showAlert('User successfully saved to application', 'info');
```
Comment (gautamguptabasant) [inline src/Pages/User.js]: nit: User successfully added

File: src/components/ApplnBar.js [hunk 2]
```diff
@@ -23,27 +27,40 @@ import CropInfoPage from 'Pages/CropInfoPage';
 import CropCycleInfoPage from 'Pages/CropCycleInfoPage';
 import FarmTimeline from 'Pages/FarmTImeLine';
 import React, { useState } from 'react';
+import Home from 'Pages/Home';
+import User from 'Pages/User';
+import { GoogleLogout } from 'react-google-login';
+import GoogleLogin from 'react-google-login';
+import { refreshTokenSetup } from '../utils/refreshToken';
```
Comment (gautamguptabasant) [inline src/components/ApplnBar.js]: We have absolute path imports enabled. Please update.

File: src/components/ApplnBar.js [hunk 3]
```diff
@@ -73,6 +90,8 @@ const ApplnBar = () => {
     >
       <List>
         {[
+          'USER',
```
Comment (gautamguptabasant) [inline src/components/ApplnBar.js]: Let's move this to the end of the list as it won't be the most commonly used feature.

File: src/components/ApplnBar.js [hunk 4]
```diff
@@ -82,6 +101,8 @@ const ApplnBar = () => {
           'ACTIVITY',
           '',
           'VEHICLE',
+          '',
+          'HOME',
```
Comment (gautamguptabasant) [inline src/components/ApplnBar.js]: Can move this to the top of the list.

File: src/components/ApplnBar.js [hunk 5]
```diff
@@ -123,9 +144,67 @@ const ApplnBar = () => {
...
       return <AddVehicle />;
+    } else if (page === 'USER') {
+      return <User />;
+    } else if (page === 'HOME') {
+      return <Home />;
     }
   };
 
+  const onLoginSuccess = (res) => {
+    console.log('Login Success: currentUser:', res.profileObj);
```
Comment (gautamguptabasant) [inline src/components/ApplnBar.js]: Please remove this

File: src/components/ApplnBar.js [hunk 6]
```diff
@@ -123,9 +144,67 @@ const ApplnBar = () => {
...
+    setImgUrl(res.profileObj.imageUrl);
+    setAvatarVisible('visible');
+    setLoginVisible(false);
+    setLogoutVisible('visible');
+    refreshTokenSetup(res);
+    setUserName(res.profileObj.name);
+    setEmail(res.profileObj.email);
+    localStorage.setItem('authToken', res.tokenId);
+    refreshTokenSetup(res);
```
Comment (gautamguptabasant) [inline src/components/ApplnBar.js:258]: Do we need this twice?
If yes, can you please add comments around each of them explaining what each of them does.

File: src/components/ApplnBar.js [hunk 7]
```diff
@@ -123,9 +144,67 @@ const ApplnBar = () => {
...
+  };
+
+  const onLoginFailure = (res) => {
+    console.log('Login failed: res:', res);
+    showAlert(`Failed to login`, 'error');
+  };
+
+  const onLogoutSuccess = (res) => {
+    localStorage.setItem('authToken', null);
+    setAnchorEl(null);
```
Comment (gautamguptabasant) [inline src/components/ApplnBar.js]: What Anchor element? Please use a better name to convey the same.

File: src/components/ApplnBar.js [hunk 8]
```diff
@@ -123,9 +144,67 @@ const ApplnBar = () => {
...
+
+  const onLogoutFailure = (res) => {
+    console.log('Log out Failed');
+    alert('Log out Failed');
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  const handleClick = (event) => {
```
Comment (gautamguptabasant) [inline src/components/ApplnBar.js]: There are multiple things in this component now. Please mention click of what. Same for close.

File: src/components/ApplnBar.js [hunk 9]
```diff
@@ -151,8 +230,104 @@ const ApplnBar = () => {
           <Typography variant='h6' className={classes.title}>
             Agri-Man
           </Typography>
-          <Button color='inherit'>Login</Button>
+          {loginVisible ? (
```
Comment (gautamguptabasant) [inline src/components/ApplnBar.js:328]: Please give it a shot to see if this whole thing could be a separate component to keep this file cleaner.

File: src/components/ApplnBar.js [hunk 10]
```diff
@@ -151,8 +230,104 @@ const ApplnBar = () => {
...
+            <Box>
+              <GoogleLogin
+                clientId='765498256947-104ng32fscf55ntmdc0p2hn0v6g6k0nu.apps.googleusercontent.com'
```
Comment (gautamguptabasant) [inline src/components/ApplnBar.js]: Please move this to a constant file and should not check it in git.

File: src/dataclients/UserClient.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import { API_URL, handleErrors } from 'utils/FetchUtils';
+
+/**
+ * API to create a new field
+ * @param {object} payload
+ */
+export const createUser = async (payload) => {
```
Comment (gautamguptabasant) [inline src/dataclients/UserClient.js:7]: Can you please take up deletion of user as well as the next change. That also falls under bare minimum I think :)

**Review by gautamguptabasant — CHANGES_REQUESTED** [review]
Great job on this overall.

File: src/components/ApplnBar.js (hunk 9 above)
Comment (mahimavuppuluri) [reply src/components/ApplnBar.js:328]: will do, in next PR

**Review by gautamguptabasant — APPROVED** [review]

**Pull request merged**